server/index.js

```diff
@@ -41,17 +43,33 @@ app.post('/reservation', function (req, res) {
   var booking = req.body;
 
   Reservation.make(booking)
-  .then((notification) => {
-    res.write(JSON.stringify(notification));
-    res.end();
-  })
-  .catch((err) => {
-    console.log('Error occurred: ', err);
-    res.status(500).send(new Error(err));
-    res.end();
-  });
+    .then((notification) => {
+      res.write(JSON.stringify(notification));
+      res.end();
+    })
+    .catch((err) => {
+      console.log('Error occurred: ', err);
+      res.status(500).send(new Error(err));
+      res.end();
+    });
 });
 
+//!! RON'S ADDITION FOR THE PUT REQUEST TO UPDATE THE RESERVATION
+//PUT is update: mongo _id field is included in the object res, use
+//Maybe use:
+//https://docs.mongodb.com/manual/reference/method/db.collection.replaceOne/
+//or https://docs.mongodb.com/manual/reference/method/db.collection.update/
+//db.collection.replaceOne(), or db.collection.update()
+app.put('/reservation', function (req, res) {
+  const booking = req.body;
+  //const time = req.params.restaurant_time;
+  // res.send(booking)
+  Reservation.updateReservation(booking);
```

Comment:
your updateReservation function takes in two parameters (see line 136 in Reservation.js) with 'err' being the first one and 'booking' being the second. be careful to keep this consistent. i'd suggest removing 'err' from your updateReservation function